### PR TITLE
Issues/2484 support validation for string object in collection

### PR DESCRIPTION
**Pull request by btratodos**
please review #2484

### CODE REVIEW

File: source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst [hunk 1]
```diff
@@ -135,15 +135,12 @@ An example of adding libraries which are required when running Bean Validation 1
 Single item check
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-For the implementation of single item check,
+For the implementation of single item check following points are necessary,
```
Comment (btratodos) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: This remaining text was not translated. No issue was raised for this.

Comment (btratodos) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: This remaining text was not translated. No issue was raised for this.

File: source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst [hunk 2]
```diff
@@ -2227,6 +2224,8 @@ This guideline classifies the definition as follows.
 
 When ValidationMessages.properties is not provided, \ :ref:`Default messages provided by Hibernate Validator<Validation_default_message_in_hibernate_validator>`\  is used.
 
+By working with \ ``MessageSource``\, you can handle Japanese messages directly without Native to Ascii.
+For details,please refer \ :ref:`Validation_without_native2ascii`
```
Comment (btratodos) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: No issue was raised for this translated text.

File: source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst [hunk 3]
```diff
@@ -2271,7 +2270,6 @@ It is explained below using the following form used at the beginning of \ :ref:`
     javax.validation.constraints.NotNull.message=is required.
     # (1)
     javax.validation.constraints.Size.message=size is not in the range {min} through {max}.
-    # (2)
```
Comment (btratodos) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: No issue was raised for this.

File: source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst [hunk 4]
```diff
@@ -2285,8 +2283,6 @@ It is explained below using the following form used at the beginning of \ :ref:`
        - Description
      * - | (1)
        - | It is possible to embed the value of attributes specified in the annotation using \ ``{Attribute name}``\ .
-     * - | (2)
-       - | It is possible to embed the invalid value using \ ``{value}``\ .
```
Comment (btratodos) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: No issue was raised for this.

File: source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst [hunk 5]
```diff
@@ -2328,6 +2324,8 @@ To display an appropriate field name, it should be defined in \ **application-me
 .. code-block:: properties
 
   form property name=field name to be displayed
+  
+It is good to define field names in the form.
```
Comment (btratodos) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: No issue was raised for this.

File: source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst [hunk 6]
```diff
@@ -2372,7 +2370,6 @@ Error messages are changed as follows.
      .. list-table::
         :header-rows: 1
         :widths: 10 90
-        :class: longtable
```
Comment (btratodos) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: No issue was raised for this.

File: source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst [hunk 7]
```diff
@@ -2391,7 +2388,33 @@ Error messages are changed as follows.
...
+        		:class: longtable
+
+        		* - Sr. No.
+				  - Description
+        		* - | (1)
 
+          		  - Messages that are actually generated from the above message definition are embedded in \ ``${validatedValue}``\ with values entered in the form.
+            		If confidential information is included in the input value, be careful not to use \ ``${validatedValue}``\ so that confidential information is not displayed in the message.
+
+            		For details, please refer,\ `Hibernate Validator Reference Guide(Interpolation with message expressions) <http://docs.jboss.org/hibernate/validator/5.2/reference/en-US/html/ch04.html#section-interpolation-with-message-expressions>`_\.
```
Comment (btratodos) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: For the above block of lines, no issue was raised.

File: source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst [hunk 8]
```diff
@@ -2733,7 +3185,7 @@ Creation of Bean Validation annotation by implementing new rules
 
 Any rule can be created by implementing \ ``javax.validation.ConstraintValidator``\  interface and creating annotation that uses this Validator.
 
-The method of usage is as follows.
+The following 3 ways can be cited as applications.
```
Comment (btratodos) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: Modified the English translation as per Japanese text.No issue was raised for this.

File: source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst [hunk 9]
```diff
@@ -3686,6 +4138,15 @@ An example to show how to create an exception handling class is given below.
...
 
+.. warning::
+
+    An error message can occur by using the  \ ``ConstraintViolation#getMessage``\ method, but since the message completion by Spring's function is not done, field names in error message at \ ``{0}``\ cannot be embedded.
+    
+    Instead, field names can be obtained with the \ ``ConstraintViolation#getPropertyPath``\ method.
+    
+    For message completion by Spring's function, please see the note of:ref:`Validation_message_in_validationmessages`.
+    
+    For details about \ ``ConstraintViolation``\,please refer \ `Hibernate Validator reference <http://docs.jboss.org/hibernate/validator/5.2/reference/en-US/html_single/#section-constraint-violation-methods>`_\.
```
Comment (btratodos) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: No issues was raised for this block of lines

File: source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst [hunk 10]
```diff
@@ -3894,7 +4355,8 @@ Refer to \ `Hibernate Validator specifications <http://docs.jboss.org/hibernate/
 
    * - \ ``@URL``\
      - It is applicable to any implementation class of \ ``CharSequence``\  interface
-     - Validate whether it is compliant with RFC2396.
+     - Verify that it is valid as a URL. String validation is performed using the constructor of \ ``java.net.URL``\
+       The protocol that is valid as the URL depends on the protocol supported by the JVM (\ ``http``\ ,\ ``https``\ ,\ ``file``\ ,\ ``jar``\ etc)
```
Comment (btratodos) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: No issue was raised for this lines.

File: source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst [hunk 11]
```diff
@@ -3917,27 +4379,21 @@ Refer to \ `Hibernate Validator specifications <http://docs.jboss.org/hibernate/
...
-    * \ ``@LuhnCheck``\
-    * \ ``@Mod10Check``\
-    * \ ``@Mod11Check``\
-    * \ ``@ModCheck``\ (deprecated API from 5.1.0.Final)
+.. tip::
 
-    This bug occurs because of the flaws in message definitions provided by default,
-    and it is possible to avoid them by overwriting the default messages by an appropriate message.
+     If you wish to validate protocols that JVM does not support in \ ``@URL``\, use \ ``org.hibernate.validator.constraintvalidators.RegexpURLValidator``\ provided by Hibernate .
+     This class is a Validator class corresponding to the \ ``@URL``\ annotation, it verifies the URL format by regular expression, and it can also validate the protocol which JVM does not support.
```
Comment (btratodos) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: No issue was raised for this lines.

File: source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst [hunk 12]
```diff
@@ -3917,27 +4379,21 @@ Refer to \ `Hibernate Validator specifications <http://docs.jboss.org/hibernate/
...
 
-    In case of overwriting the default messages,
-    it is advisable to define an appropriate message 
-    by creating :file:`ValidationMessages.properties` directly under the class path (normal src/main/resources).
+     * If the check rule of \ ``@URL``\  has to be changed for whole apllication then,
+       in XML change the Validator class to \ ``RegexpURLValidator``\ as described in \ `JavaDoc <https://docs.jboss.org/hibernate/validator/5.2/api/org/hibernate/validator/constraints/URL.html>`_\
+     * Apply regular expression verification only to some items,if default rule is to be used,\ ``@URL``\ creates new annotation and \ ``javax.validation.ConstraintValidator``\ implementation class that does the same verification as \ ``RegexpURLValidator``\, 
+       apply verification for the necessary items by the created annotations.
```
Comment (btratodos) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: No issue was raised for this lines.

File: source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst [hunk 13]
```diff
@@ -3917,27 +4379,21 @@ Refer to \ `Hibernate Validator specifications <http://docs.jboss.org/hibernate/
...
 
-    For appropriate message definition, refer to:
-    \ `Modifications for Hibernate Validator 5.2 version (next minor version upgrade) <https://github.com/hibernate/hibernate-validator/commit/5a9d7bae26bccb15229ae5612d67506a7a775b48#diff-762e02c90cfb2f00b0b2788486e3fd5e>`_\ .
+     Apply according to the application etc.
 
+     For details on changing check rules in XML, please refer \ `Hibernateのrefernence <https://docs.jboss.org/hibernate/validator/5.2/reference/en-US/html/ch07.html#section-configuration-validation-xml>`_\,
+     and about how to create new annotations please refer, \ :ref:`Validation_implement_new_constraint`\
+
```
Comment (btratodos) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: No issue was raised for this lines.

File: source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst [hunk 14]
```diff
@@ -4219,6 +4675,10 @@ Add a dependent library for the rules that are to be used. An example for how to
         </dependency>
     </dependencies>
 
+.. note::
+
+    In the above setting example,since it is a premise to manage the dependent library version with the parent project terasoluna-gfw-parent, specifying the version in pom.xml is not necessary.
+
```
Comment (btratodos) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: No issue was raised for this lines.

File: source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst (hunk 1 above)
Comment (asaharas) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: thanks.
I think the sentence is missed at previous release timing

File: source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst (hunk 2 above)
Comment (asaharas) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: the sentence is removed in #2311

File: source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst (hunk 3 above)
Comment (asaharas) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: the sentence is added in #2311

File: source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst (hunk 4 above)
Comment (asaharas) [inline source_en/ArchitectureInDetail/WebApplicationDetail/Validation.rst]: the sentence is added in #2311

**Comment by asaharas** [comment]
I have checked all comments.
file changes are OK.

**Pull request merged**